server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -801,8 +802,32 @@ def formula(
         #   [my age] + 1 / 2
         # into
         #   CAST("my age" AS float) + 1 / 2
+        # TODO : an AST should be implemented here to sanitize the formula
+
+        def _sanitize_formula(formula):
+            # we replace [ to "
+            formula = re.sub(r'[\[\]]', self.QUOTE_CHAR, formula)
+
+            # we add quotes on columns for postgresql
+            formula = re.sub(
+                r'([a-zA-Z_a-zA-Z]+)', r'{}\1{}'.format(self.QUOTE_CHAR, self.QUOTE_CHAR), formula
+            ).replace('""', self.QUOTE_CHAR)
 
-        query = query.select(LiteralValue(handle_zero_division(step.formula)).as_(step.new_column))
+            if '/' in formula:
+                formula = re.sub(
+                    r'(?<=/)\s*(\w+)|(?<=/)\s*(\"?.*\"?)', r' NULLIF(\1\2, 0)', formula
+                )
+            if '%' in formula:
+                formula = re.sub(
+                    r'(?<=%)\s*(\w+)|(?<=%)\s*(\"?.*\"?)', r' NULLIF(\1\2, 0)', formula
+                )
+
+            return formula
+
+        step.formula = _sanitize_formula(step.formula)
+        query: "QueryBuilder" = self.QUERY_CLS.from_(prev_step_name).select(
+            *columns, LiteralValue(step.formula).as_(step.new_column)
+        )
```

Comment:
```suggestion
        formula = _sanitize_formula(step.formula)
        query: "QueryBuilder" = self.QUERY_CLS.from_(prev_step_name).select(
            *columns, LiteralValue(formula).as_(step.new_column)
        )
```

Don't mutate the pipeline

Reply:
https://github.com/ToucanToco/weaverbird/pull/1425/commits/cf01b697aa9209da620bdf02edca5f1697a75cd3